ext/reflection/php_reflection.c

```diff
@@ -5715,6 +5715,106 @@ ZEND_METHOD(reflection_property, hasType)
 }
 /* }}} */
 
+/* {{{ proto public bool ReflectionProperty::hasDefaultValue()
+   Returns whether property has a default value */
+ZEND_METHOD(reflection_property, hasDefaultValue)
+{
+	reflection_object *intern;
+	property_reference *ref;
+	zend_property_info *prop_info;
+	zval *prop, prop_copy;
+	zend_class_entry *ce;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	GET_REFLECTION_OBJECT_PTR(ref);
+
+	prop_info = ref->prop;
+	ce = prop_info->ce;
+
+	prop = NULL;
+	if ((prop_info->flags & ZEND_ACC_STATIC) != 0) {
+		prop = &ce->default_static_members_table[prop_info->offset];
+		ZVAL_DEINDIRECT(prop);
+	} else if ((prop_info->flags & ZEND_ACC_STATIC) == 0) {
```

Comment:
Can't it be just an `else`? Assuming that `&` in the `if` clause return either true or false, not other possible values?

Reply:
I agree - this should be:
```suggestion
	} else {
```

Reply:
This is still unresolved.